src/commands/pull.js

```diff
@@ -0,0 +1,98 @@
+const AdmZip = require('adm-zip');
+const semverGt = require('semver/functions/gt');
+const { get } = require('axios');
+const { ApifyCommand } = require('../lib/apify_command');
+const { success, error } = require('../lib/outputs');
+const { getLoggedClientOrThrow } = require('../lib/utils');
+
+class PullCommand extends ApifyCommand {
+    async run() {
+        const { args } = this.parse(PullCommand);
+        const apifyClient = await getLoggedClientOrThrow();
+        const cwd = process.cwd();
+
+        const { actorId } = args;
+        const actor = await apifyClient.actor(actorId).get();
+        if (!actor) throw new Error(`Cannot find actor with ID '${actorId}' in your account.`);
+        const { name, versions } = actor;
+
+        const latestVersion = versions.reduce((match, curr) => {
+            if (semverGt(`${curr.versionNumber}.0`, `${match.versionNumber}.0`)) return curr;
+            return match;
+        });
+
+        const dirpath = path.join(cwd, name);
+        mkdirp.sync(dirpath);
+
+        if (!(fs.readdirSync(dirpath).length === 0)) {
+            error(`Directory ${dirpath} is not empty. Please empty it or choose another directory.`);
+        }
+
+        let isPullSuccessful = false;
+
+        switch (latestVersion.sourceType) {
+            case 'TARBALL': {
+                const { tarballUrl } = latestVersion;
+
+                const { data } = await get(tarballUrl, { responseType: 'arraybuffer' });
+
+                const zipFile = new AdmZip(Buffer.from(data, 'binary'));
+                zipFile.extractAllTo(dirpath);
+                break;
+            }
+            case 'SOURCE_FILES': {
+                const { sourceFiles } = latestVersion;
+                for (const file of sourceFiles) {
+                    if (!file.folder) {
+                        mkdirp.sync(dirpath);
+                        fs.writeFileSync(`${dirpath}/${file.name}`, file.content);
+                    }
+                }
+                isPullSuccessful = true;
+                break;
+            }
+            case 'GIT_REPO': {
+                // e.g. https://github.com/jakubbalada/Datasety.git#master:RejstrikPolitickychStran
+                const { gitRepoUrl } = latestVersion;
+                const [repoUrl, branchDirPart] = gitRepoUrl.split('#');
+
+                let branch;
+                let dir;
+                if (branchDirPart) [branch, dir] = branchDirPart.split(':');
+
+                const emitter = tiged(branchDirPart ? `${repoUrl}/${dir}#${branch}` : repoUrl);
+                try {
+                    await emitter.clone(dirpath);
+                } catch (err) {
+                    error(`Failed to pull actor from ${gitRepoUrl}.`);
+                    break;
+                }
+
+                isPullSuccessful = true;
+                break;
+            }
+            case 'GITHUB_GIST':
+                throw new Error('Pulling from GitHub Gist is not supported.');
```

Comment:
This should be really easy to do, a gist is just an archive, so it can use the same code as `TARBALL`, just with the right URL.

Reply:
Well, maybe easy, but 1. almost nobody using gists, 2. not even required in specs. I'd rather skip this.

Reply:
I think the specs just forgot about this, I really don't see why can't we support it